shell/src/main/java/org/apache/accumulo/shell/commands/DUCommand.java

```diff
@@ -80,13 +109,53 @@ public int execute(final String fullCommand, final CommandLine cl, final Shell s
     }
 
     try {
-      String valueFormat = prettyPrint ? "%9s" : "%,24d";
-      for (DiskUsage usage : shellState.getAccumuloClient().tableOperations()
-          .getDiskUsage(tables)) {
-        Object value = prettyPrint ? NumUtil.bigNumberForSize(usage.getUsage()) : usage.getUsage();
-        shellState.getWriter()
-            .println(String.format(valueFormat + " %s", value, usage.getTables()));
+      final String valueFormat = prettyPrint ? "%s" : "%,d";
+      final TableDiskUsageResult usageResult = shellState.getAccumuloClient().tableOperations()
+          .getEstimatedDiskUsage(tables, verbose, auths);
+
+      // Print results for each table queried
+      for (Map.Entry<TableId,AtomicLong> entry : usageResult.getTableUsages().entrySet()) {
+        final Object usage =
+            prettyPrint ? NumUtil.bigNumberForSize(entry.getValue().get()) : entry.getValue().get();
+
+        // Print the usage for the table
+        shellState.getWriter().print(String.format("%s  %s  used total: " + valueFormat,
+            shellState.getContext().getTableName(entry.getKey()), entry.getKey(), usage));
+
+        // Print usage for all volume(s) that contain files for this table
+        Optional.ofNullable(usageResult.getVolumeUsages().get(entry.getKey()))
+            .ifPresent(tableVolUsage -> tableVolUsage.entrySet()
+                .forEach(vuEntry -> shellState.getWriter()
+                    .print(String.format("  %s: " + valueFormat, vuEntry.getKey(),
+                        prettyPrint ? NumUtil.bigNumberForSize(vuEntry.getValue().get())
+                            : vuEntry.getValue().get()))));
+
+        // Check/print if this table has any shared files
+        final Optional<Set<TableId>> sharedTables =
+            Optional.ofNullable(usageResult.getSharedTables().get(entry.getKey()));
+        final boolean hasShared = sharedTables.map(st -> st.size() > 0).orElse(false);
+
+        shellState.getWriter().print(String.format("  has_shared: %s", hasShared));
+        if (hasShared) {
+          shellState.getWriter().print(String.format("  shared with:  %s", sharedTables.get()));
+        }
+        shellState.getWriter().println();
```

Comment:
I don't think it's necessary to change this shell command at all. The only differences should be in the API method's implementation, and maybe this command's description to emphasize it's an estimate. None of the printing of the output needs to be changed at all to achieve the desired goal of this PR, to get size info from the metadata.